test/features/prebooking.js

```diff
@@ -0,0 +1,303 @@
+      cid_id: '456'
+    }]
+  };
+
+  Scenario('Consider Today: 7am - 6:59am', () => {
+    Scenario('New Valid Pre-bookings replace existing pre-bookings', () => {
+
+      var followingPayload = {
+        Output: [{
+          timestamp: validTimestamp,
+          location: 'smallone female holding',
+          task_force: 'ops1',
+          cid_id: '111'
+        }, {
+          timestamp: validTimestamp,
+          location: 'smallone female holding',
+          task_force: 'ops1',
+          cid_id: '444'
+        }, {
+          timestamp: validTimestamp,
+          location: 'bigone male holding',
+          task_force: 'ops1',
+          cid_id: '222'
+        }, {
+          timestamp: validTimestamp,
+          location: 'bigone male holding',
+          task_force: 'ops1',
+          cid_id: '333'
+        }]
+      };
+      var expectedFemaleCentre = 'smallone';
+      var expectedMaleCentre = 'bigone';
+
+      before(function () {
+        global.testConfig.initializeBarrelsFixtures = false;
+        return global.initializeBarrelsFixtures();
+      });
+
+      after(function () {
+        global.testConfig.initializeBarrelsFixtures = true;
+      });
+
+      Given(`a prebooking with cid id "${payload.Output[0].cid_id}" has already occurred`, () =>
+        createRequest(payload, '/depmu_entry/prebooking').then(
+          () => findPrebookingByCID(payload.Output[0].cid_id).then((models) => {
+            expect(models.length).to.equal(1);
+            expect(models[0].cid_id).to.equal(parseInt(payload.Output[0].cid_id));
+          })
+        )
+      );
+
+      When(`new valid prebookings occur`, () => createRequest(followingPayload, '/depmu_entry/prebooking'));
+
+      Then('previous prebookings recorded are removed',
+        () => findPrebookingByCID(payload.Output[0].cid_id).then((models) => expect(models.length).to.equal(0))
+      );
+
+      And(`new female prebookings are created`, () =>
+        Prebooking.find({where: {gender: 'female'}}).then((models) => expect(models.length).to.equal(2))
+      );
+      And(`new male prebookings are created`, () =>
+        Prebooking.find({where: {gender: 'male'}}).then((models) => expect(models.length).to.equal(2))
+      );
+
+    });
+
+    Scenario('New Invalid Pre-bookings are ignored', () => {
+      var followingPayload = {
+        Output: [{
+          timestamp: validTimestamp,
+          location: 'smallone female holding',
+          cid_id: '111'
+        }]
+      };
+
+      before(function () {
+        global.testConfig.initializeBarrelsFixtures = false;
+        return global.initializeBarrelsFixtures();
+      });
+
+      after(function () {
+        global.testConfig.initializeBarrelsFixtures = true;
+      });
+
+      Given(`a prebooking with cid id "${payload.Output[0].cid_id}" has already occurred`, () =>
+        createRequest(payload, '/depmu_entry/prebooking').then(
+          () => findPrebookingByCID(payload.Output[0].cid_id).then((models) => {
```

Comment:
same as above, new line for `.then` and i think you're putting a .then inside a .then